File: netlify/functions/check.py
from credify_common import json_response, parse_body, run_check_core


def handler(event, context):
    data = parse_body(event)
    question = (data.get("question") or "").strip()
    mode = data.get("mode") or "ask"
    deep = bool(data.get("deep"))
    if not question:
        return json_response(400, {"error": "No question provided."})
    return json_response(200, run_check_core(question, mode, deep))

File: netlify/functions/check.js
import { runCheckCore } from "./lib/providers.js";

const json = (status, payload) =>
  new Response(JSON.stringify(payload), { status, headers: { "content-type": "application/json" } });

export default async (req) => {
  if (req.method !== "POST") return json(405, { error: "Method not allowed." });

  let data;
  try {
    data = await req.json();
  } catch {
    data = {};
  }
  const question = (data.question || "").trim();
  const mode = data.mode || "ask";
  const deep = Boolean(data.deep);
  if (!question) return json(400, { error: "No question provided." });

  const result = await runCheckCore(question, mode, deep);
  return json(200, result);
};

export const config = { path: "/api/check" };
